t/ui/27-plugin_obs_rsync_obs_status.t

```diff
@@ -71,28 +83,95 @@ my %fake_response_by_project = (
     Proj0 => 'invalid XML',
 );
 
+my $auth_header_exact
+  = qq(Signature keyId="dummy-username",algorithm="ssh",)
+  . qq(signature="U1NIU0lHAAAAAQAAADMAAAALc3NoLWVkMjU1MTkAAAAgSKpcECPm8Vjo9UznZS+)
+  . qq(M/QLjmXXmLzoBxkIbZ8Z/oPkAAAAaVXNlIHlvdXIgZGV2ZWxvcGVyIGFjY291bnQAAAAAAAAABn)
+  . qq(NoYTUxMgAAAFMAAAALc3NoLWVkMjU1MTkAAABA8cmvTy1PgpW2XhHWxQ1yw/wPGAfT2M3CGRJ3II)
+  . qq(7uT5Orqn1a0bWlo/lEV0WiqP+pPcQdajQ4a2YGJvpfzT1uBA==",)
+  . qq (headers="(created)",created="1664187470");
+
 note 'Starting fake API server';
-my $server_instance = process sub {
+
+my $server_process = sub {
     my $mock = Mojolicious->new;
     $mock->mode('test');
+
+    my $www_authenticate = qq(Signature realm="Use your developer account",headers="(created)"); # uncoverable statement
+    $mock->routes->get(
+        '/build/ProjWithAuth/_result' => sub {
+            my $c = shift;
+
+            if ($c->req->headers->authorization) {
+                return $c->render(status => 200, text => $fake_response_by_project{Proj1});
+            }
+
+            $c->res->headers->www_authenticate($www_authenticate);
+
+            return $c->render(status => 401, text => 'login');
+        });
+
+    $mock->routes->get(
+        '/build/ProjTestingSignature/_result' => sub {
+            my $c = shift;
+
+            my $client_auth_header = $c->req->headers->authorization // '';
+
+            if ($auth_header_exact eq $client_auth_header) {
+
+                return $c->render(status => 200, text => $fake_response_by_project{Proj1});
+            }
+
+            $c->res->headers->www_authenticate($www_authenticate);
+            return $c->render(status => 401, text => 'login');
+        });
+
     for my $project (sort keys %fake_response_by_project) {
         $mock->routes->get(
-            "/public/build/$project/_result" => sub {
+            "/build/$project/_result" => sub {
                 my $c = shift;
                 my $pkg = $c->param('package');
                 return $c->render(status => 404) if !$pkg and $project ne 'Proj1';
-                return $c->render(status => 200, text => $fake_response_by_project{$project});
+                return $c->render(status => 200, text => $fake_response_by_project{$project});   # uncoverable statement
             });
     }
     my $daemon = Mojo::Server::Daemon->new(app => $mock, listen => [$host]);
     $daemon->run;
     note 'Fake API server stopped';
-    _exit(0);
+    Devel::Cover::report() if Devel::Cover->can('report');
+    _exit(0);    # uncoverable statement
 };
+
+my $server_instance = process(
+    $server_process,
+    max_kill_attempts => 0,
+    blocking_stop => 1,
+    _default_blocking_signal => POSIX::SIGTERM,
+    kill_sleeptime => 0
+);
+
 $server_instance->set_pipes(0)->start;
 wait_for_or_bail_out { IO::Socket::INET->new(PeerAddr => '127.0.0.1', PeerPort => $port) } 'API';
 
-my ($t, $tempdir, $home, $params) = setup_obs_rsync_test(url => $url);
+my $ssh_keyfile = tempfile();
```

Comment:
Even in tests we should use descriptive tempfiles (I know that we have a lot of blank `tempfile` calls left in our tests, but we should get into the habit of adding a filename for new code)
```suggestion
my $ssh_keyfile = tempfile("$FindBin::Script-sshkey-XXXXX");
```